components/camel-olingo4/camel-olingo4-api/src/main/java/org/apache/camel/component/olingo4/api/impl/Olingo4AppImpl.java

```diff
@@ -511,6 +519,19 @@ private AbstractHttpEntity writeContent(final Edm edm, final UriInfo uriInfo, fi
             List<UriResource> listResource = uriInfo.getUriResourceParts();
             UriResourceKind lastResourceKind = listResource.get(listResource.size() - 1).getKind();
             switch (lastResourceKind) {
+            case action:
```

Comment:
@kayuma looks like the code for action and entitySet are the same, may be use common code for both?

Reply:
Done